### DISCUSSION TITLE
How to deploy nextjs and express app on  vercel server code not triggering server routes not found

### DISCUSSION THREAD

**Discussion by sunil-bagde** [vercel/next.js#13306] [Help]
Is it possible  on  hobby plan?

Current structure 

standard next app 
+
server.js  where server code lives E.g  

https://github.com/zeit/next.js/blob/canary/examples/custom-server-express/server.js

scripts 

```sh 
"scripts": {
        "dev": "nodemon index.js",
        "build": "next build",
        "start": "NODE_ENV=production node index.js"
    },

```
Server express routes not triggering

**Comment by lieutandat** [comment]
@im-sunil did you get solutions? I met the same issue as you

**Reply by sunil-bagde** [reply]
https://mobile.twitter.com/iamsunil786/status/1264624277040459781

This is solution 

You express server code have to move in new /api/ routes 

Here you can refer 
https://nextjs.org/docs/api-routes/introduction

In api/* you can write any server logic.

**Reply by weyert** [reply]
I have the same issue when you use a custom `server.js` to introduce `SSL` for local development the API routes appear to always return 404. My `server.js` is listed below, next to it I have created a `/api/quantify/page.ts` which just returns the Hello World via `response.end(JSON.stringify({ name: 'Hello World' }))`. I am not sure what I am doing incorrectly, what's the best practice approach for enabling a SSL certificate during development?

```
/* eslint-disable no-console */
const { createServer: createSecureServer } = require('https')
const { createServer } = require('http')

const { parse } = require('url')
const fs = require('fs')
const next = require('next')

const HOST = process.env.HOST || 'localhost'
const serverPort = process.env.PORT || 5000
const isDevelopment = process.env.NODE_ENV !== 'production'
const app = next({ dev: isDevelopment })
const handle = app.getRequestHandler()

const httpsOptions = isDevelopment
  ? {
      key: fs.readFileSync('./certs/wildcard-dev-ssl.key'),
      cert: fs.readFileSync('./certs/wildcard-dev-ssl.crt'),
    }
  : {}

app
  .prepare()
  .then(() => {
    const secureFunction = isDevelopment ? createSecureServer : createServer
    secureFunction(httpsOptions, (req, res) => {
      const parsedUrl = parse(req.url, true)
      handle(req, res, parsedUrl)
    }).listen(serverPort, err => {
      if (err) throw err
      console.log(`> Ready on https://${HOST}:${serverPort} NODE_ENV: ${process.env.NODE_ENV || 'development'}`)
    })
  })
  .catch(ex => {
    console.error(ex.stack)
    // eslint-disable-next-line no-process-exit
    process.exit(1)
  })
```

**Reply by sunil-bagde** [reply]
Here i assuming having 2 diff servers 

1- Node server.js 
2. next js   -> next dev or next start 

next js can't communicate with you node server on vercel 

whatever you doing on backend you have to move in pages/api/* and remove a node http  express js code because next js and vercel does for you. with pages/api/*  NOTE node need to create node server 

you-base-url/api/quantify/page you can run. but its have to in local yarn dev or [ yarn start using PM2 vercel does for you]

vercel all site are https proof.

if dont mind can you share repo example?

**Reply by weyert** [reply]
Looks like it started to work after I deleted `.next` and `build` directories of my project. Can't explain it really.

**Reply by lieutandat** [reply]
> https://mobile.twitter.com/iamsunil786/status/1264624277040459781
> 
> This is solution
> 
> You express server code have to move in new /api/ routes
> 
> Here you can refer
> https://nextjs.org/docs/api-routes/introduction
> 
> In api/* you can write any server logic.

good solution, thank for your quickly reply 😃

**Reply by GiftsonGabogen** [reply]
On development you cant use ssl if your using nextjs api routes only I think, we still need to make new server to be able to use ssl
